### ISSUE TITLE
Expose InvokeOptions through the Nodejs provider interface

### ISSUE THREAD

**Issue by danielrbradley** [pulumi/pulumi#9473]
## Hello!

- Vote on this issue by adding a 👍 reaction
- If you want to implement this feature, comment to let us know (we'll work with you on design, scheduling, etc.)

## Issue details

When invoking a function via the SDK there's options for setting parent and other properties via the `opts`.

The Nodejs provider interface doesn't currently expose the `opts` to the provider implementation:

```typescript
invoke?: (token: string, inputs: any) => Promise<InvokeResult>;
```

### Affected area/feature

- NodeJS SDK -> Provider

**Comment by Frassle** [comment]
This isn't another one of those cases (like Invoke) where we've reused the protobuf message for both the ResourceMonitor and the ResourceProvider but fill in different fields for each?

**Comment by danielrbradley** [comment]
Oh actually this should be invokes not calls. Calls don't have opts because they're nested within a resource, but invokes are standalone.

> This isn't another one of those cases (like Invoke) where we've reused the protobuf message for both the ResourceMonitor and the ResourceProvider but fill in different fields for each?

I don't think so - this isn't just options in the gRPC messages - it's actually generated as options in the SDK.

**Comment by Frassle** [comment]
Right but opts we pass from the SDK to the engine, might not get passed from engine to provider.

Invokes for the provider currently look like:
```
message InvokeRequest {
    string tok = 1;                  // the function token to invoke.
    google.protobuf.Struct args = 2; // the arguments for the function invocation.

    // We used to send ResourceInvokeRequest for both provider invokes and monitor invokes, despite them being
    // different. We've now split them but need to make sure we don't confuse any old plugins/monitors making
    // sure those fields don't get reused.
    reserved 3 to 6;
    reserved "provider", "version", "acceptResources", "pluginDownloadURL";
}
```

Just a token and args struct.

**Comment by danielrbradley** [comment]
Yup, ok so it looks like this gets lost earlier than provider API - so it's a wider issue. I think @justinvp might also be aware of this as was just talking through with Paul about if invokes will actually work with creating resources.

I'm just going to remove the code that was needing this for now and just not include it.

**Comment by Frassle** [comment]
> Paul about if invokes will actually work with creating resources.

Could you pass the parent URN as an arg?

**Comment by danielrbradley** [comment]
Could do something like that as a hack, but would prefer to leave it out to avoid adding a hack to maintain. Might be worth closing this if there's something else already tracking creation of resources in invokes?

**Comment by Frassle** [comment]
> Might be worth closing this if there's something else already tracking creation of resources in invokes?

I don't think there is.

**Comment by justinvp** [comment]
#7583
